### PR TITLE
Run tests with Chrome headless

### PR DESCRIPTION
**Pull request by maclover7**
TL;DR: Drop-in code replacement for Phantom with Chrome headless.

The Google Chrome team came out with a slick JavaScript client library
called [puppeteer](https://github.com/googlechrome/puppeteer) to control
Chrome headless. This is a fairly (IMHO) straightforward port from
running QUnit tests through Phantom to instead using Chrome headless.
The functions are almost identical with the exceptions of how to
actually boot/load up the browser and page, and also the logging code
in-line.

There are a few TODOs (things before this PR is merged) and Future
Improvements (things likely out of scope for this PR) that I was able to
come up with, though I'm sure I may have missed a few things:

TODOs:

- Extract out the `addLogging` function to a package similar to `qunit-phantomjs-runner` (`qunit-chrome-runner`?): right now copy and pasting the code from the Phantom runner works, but this isn't sustainable. Not sure if the Ember core team wants to maintain a small package like this, or who will own it.
- Make sure testem is okay with removal of PhantomJS: there are a bunch of config files where Phantom is mentioned, those need to be looked at
- Figure out when to remove Phantom completely from `bin` scripts: my initial thought was to run tests through both phantom and Chrome headless for a little while to make sure nothing was loudly failing, but I'll love feedback from reviewers about that

Future Improvements:

- Eventually remove all PhantomJS code from `bin` scripts, and from all documentation/guides/etc
- Remove `REDEFINE_SUPPORTED` / `handleBrokenPhantomDefineProperty` from [ember-metal](https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/properties.js)
- Find a way to more responsibly handle browsers: I tried to setup some `browser.close()` calls to better manage the current situation, but things got messy quickly. I think this is outside of the scope of just getting this running in the first place.
- Once Chrome headless hits stable release, add it as a Travis addon: currently the puppeteer npm package ships with a copy of the Chrome canary web browser, and it isn't necessary to install Chrome again
- Finish migrating all Ember projects off of Phantom: ember-cli has already done the migration, not sure what other repos still need attention

### CODE REVIEW

**Comment by maclover7** [comment]
Looks like tests are failing because Ember runs its CI with Node 6... and also looks like puppeteer has [no plans to add support for Node 6](https://github.com/GoogleChrome/puppeteer/issues/316#event-1209293144) 😢

**Comment by xdumaine** [comment]
The readme for Puppeteer mentions that it can run on 6.4.0. Good news?

**Comment by xdumaine** [comment]
My team would love to help contribute to this effort. Is there anything specific where you could use some help?

**Comment by rwjblue** [comment]
Closing in favor of https://github.com/emberjs/ember.js/pull/15877
